mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/binarystorage/GridFSDelegator.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.mongodb.binarystorage;
+
+import java.io.ByteArrayOutputStream;
+
+import org.hibernate.engine.jdbc.BinaryStream;
+import org.hibernate.ogm.datastore.mongodb.options.impl.GridFSBucketOption;
+import org.hibernate.ogm.model.spi.Tuple;
+import org.hibernate.ogm.options.spi.OptionsContext;
+
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.gridfs.GridFSBucket;
+import com.mongodb.client.gridfs.GridFSBuckets;
+import org.bson.Document;
+import org.bson.types.ObjectId;
+
+/**
+ * The implementation provides features for GridFS usage
+ *
+ * @see <a href ="https://docs.mongodb.com/manual/core/gridfs">GridFS documentation</a>
+ * @author Sergey Chernolyas &amp;sergey_chernolyas@gmail.com&amp;
+ */
+public class GridFSDelegator implements BinaryStorageDelegator {
+
+	/**
+	 * store binary content of field to GridFS
+	 *
+	 * @param mongoDatabase current instance of MongoDB
+	 * @param optionsContext options context
+	 * @param currentDocument current BSON document
+	 * @param fieldName fieldName with binary content
+	 * @param tuple tuple for processing
+	 */
+
+	@Override
+	public void storeContentToBinaryStorage(MongoDatabase mongoDatabase,OptionsContext optionsContext, Document currentDocument,String fieldName,
+			Tuple tuple ) {
+		String gridfsBucketName = optionsContext.getUnique( GridFSBucketOption.class );
+
+		GridFSBucket gridFSFilesBucket = getGridFSFilesBucket( mongoDatabase, gridfsBucketName );
+		String fileName = "";
+		BinaryStream binaryStream = currentDocument.get( fieldName,BinaryStream.class );
+		ObjectId uploadId = gridFSFilesBucket.uploadFromStream( fileName, binaryStream.getInputStream() );
+		//change value of the field (BinaryStream -> ObjectId)
```

Comment:
This comment is not particularly helpful, you are just repeating what the method does.